query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PromQLQuery.java

```diff
@@ -0,0 +1,39 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Singular;
+
+@Getter
+@Builder(builderClassName = "Builder", buildMethodName = "build")
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class PromQLQuery {
+
+  @NonNull @Singular private List<String> queries;
+  private boolean isInstantRequest;
+  private Instant startTime;
+  private Instant endTime;
+  private Duration step;
+
+  static class Builder {
```

Comment:
@rish691 Let me know if you think, we want to divide this into two Query objects - one for Instant and another for RangeQuery.

Reply:
A boolean that determines behavior sounds like it should be two objects. Guessing only one of the times is used by an instant query, but it's not clear which one (edit: looking below looks like its end time, which requires reading the impl)?

Although I guess the broader question I have - what do instant queries mean in the context of our current query language? I probably missed this conversation, but I'm curious if this is actually required for parity.

Reply:
>Although I guess the broader question I have - what do instant queries mean in the context of our current query language?

Aggregation metric query without `dataTimeConvert` group by clause (non-time series query), but asking for metric value for a given point in time. e.g error_count values on last 15 mins on the metric page (above time series, we are showing Total). Query Service example in description note.

e.g PQL query:
```
Select service_id, sum(numCalls) 
FROM spanEventView 
WHERE tenant_id = '__default' 
AND ( service_id != 'null' 
AND ( start_time_millis >= 1631175154903 AND start_time_millis < 1631178754903 ) 
AND ( service_id = '27857c9c-f6fa-30fe-ac1d-48fd07374775' ) ) 
GROUP BY service_id 
limit 1
```

Reply:
Discussed with @rish691, breaking PromQLQuery object into two PromQLInstantQuery and PromQLRangeQuery.

Reply:
Done